### PR TITLE
Feat/notes

### PR DESCRIPTION
**Pull request by DaniilVolobuiev**
## Describe your changes

>feat: Notes module. Endpoints for create and get notes. Endpoint to download a file. While creating a note it is possible to attach file. Get notes have sorting and searching functionality.
> DB changes. Added note and file tables. They have one-to-one relationship. 
>DB changes. Doctor has one-to-many to note. Note has many-to-one to doctor.
>DB changes. Patient has one-to-many to note. Note has many-to-one to patient.  
Note
image
 File
image

>DB changes. Doctor has one-to-many to note. Note has many-to-one to doctor.  
> TODO patient is hardcoded. Waiting for patient list to be ready.

## Issue ticket code (and/or) and link

> [Link to JIRA ticket](https://smartdc23.atlassian.net/browse/SMAR-81) 81

### General

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions is public only if it's used outside the class
- [] No hardcoded values. Waiting for patient list. 
- [x] check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### CODE REVIEW

File: src/modules/notes/notes.controller.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+@ApiTags('Notes')
+export default class NotesController {
+  constructor(private notesService: NotesService) {}
+
+  // get all notes
+
+  @ApiOperation({ summary: 'Getting All Notes' })
+  @ApiResponse({ status: 200, type: [Note] })
+  @Get('/all')
+  @UseGuards(JwtAuthGuard)
```
Comment (lyudad) [inline src/modules/notes/notes.controller.ts]: I think you need setup guard to full controller. no?

File: src/modules/notes/notes.service.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  async findAll(
+    query: Query,
+  ): Promise<{ notes: Note[]; count: number; countWithoutAnyParams: number }> {
+    try {
+      const qb = this.notesRepository.createQueryBuilder('notes');
+      const sortOrder = query.sortOrder === 'desc' ? 'DESC' : 'ASC';
+      let sortBy: string;
+      if (query.sortBy === 'Date') sortBy = 'createdAt';
```
Comment (lyudad) [inline src/modules/notes/notes.service.ts]: compare with constants please

File: src/modules/notes/notes.controller.ts (hunk 1 above)
Comment (DaniilVolobuiev) [reply src/modules/notes/notes.controller.ts]: applied guards to full controller

File: src/modules/notes/notes.service.ts (hunk 2 above)
Comment (DaniilVolobuiev) [reply src/modules/notes/notes.service.ts]: added variables

File: src/modules/notes/notes.service.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+          .insert()
+          .into(File)
+          .values({
+            filename: file.filename,
+            originalName: file.originalname,
+            size: file.size,
+            mimetype: file.mimetype,
+            note: newNote.generatedMaps[0].id,
+          })
+          .execute();
```
Comment (lyudad) [inline src/modules/notes/notes.service.ts:50]: use transaction

Comment (DaniilVolobuiev) [reply src/modules/notes/notes.service.ts:50]: added transaction

**Review by lyudad — APPROVED** [review]

**Pull request merged**